tests/moe/test_unified_moe.py

```
@@ -1085,28 +1085,51 @@ def test_trtllm_fp4_situ_rejected_on_rubin(
         assert runner.check_support() is None
 
+    def test_cute_dsl_w4a8_rejected_on_rubin(self, monkeypatch):
+        import flashinfer.utils as utils
+
+        runner = CuteDslRunner.__new__(CuteDslRunner)
+        runner.config = self._nvfp4_swiglu(
+            quant=QuantConfig(variant=QuantVariant.MXFP4)
+        )
+        runner.device = torch.device("cuda")
+        monkeypatch.setattr(utils, "get_compute_capability", lambda _: (10, 7))
+        with pytest.raises(NotImplementedError, match="W4A8.*SM107"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)